moveit_configs_utils/moveit_configs_utils/launches.py

```diff
@@ -231,3 +235,103 @@ def generate_move_group_launch(moveit_config):
         additional_env={"DISPLAY": ":0"},
     )
     return ld
+
+
+def generate_demo_launch(moveit_config):
+    """
+    Launches a self contained demo
+
+    Includes
+     * static_virtual_joint_tfs
+     * robot_state_publisher
+     * move_group
+     * moveit_rviz
+     * warehouse_db (optional)
+     * ros2_control_node + controller spawners
+    """
+    ld = LaunchDescription()
+    ld.add_action(
+        DeclareBooleanLaunchArg(
+            "db",
+            default_value=False,
+            description="By default, we do not start a database (it can be large)",
+        )
+    )
+    ld.add_action(
+        DeclareBooleanLaunchArg(
+            "debug",
+            default_value=False,
+            description="By default, we are not in debug mode",
+        )
+    )
+    ld.add_action(DeclareBooleanLaunchArg("use_rviz", default_value=True))
+
+    # If there are virtual joints, broadcast static tf by including virtual_joints launch
+    virtual_joints_launch = (
+        moveit_config.package_path / "launch/static_virtual_joint_tfs.launch.py"
+    )
+    if virtual_joints_launch.exists():
+        ld.add_action(
+            IncludeLaunchDescription(
+                PythonLaunchDescriptionSource(str(virtual_joints_launch)),
+            )
+        )
+
+    # Given the published joint states, publish tf for the robot links
+    ld.add_action(
+        IncludeLaunchDescription(
+            PythonLaunchDescriptionSource(
+                str(moveit_config.package_path / "launch/rsp.launch.py")
+            ),
+        )
+    )
+
+    ld.add_action(
+        IncludeLaunchDescription(
+            PythonLaunchDescriptionSource(
+                str(moveit_config.package_path / "launch/move_group.launch.py")
+            ),
+        )
+    )
+
+    # Run Rviz and load the default config to see the state of the move_group node
+    ld.add_action(
+        IncludeLaunchDescription(
+            PythonLaunchDescriptionSource(
+                str(moveit_config.package_path / "launch/moveit_rviz.launch.py")
+            ),
+            condition=IfCondition(LaunchConfiguration("use_rviz")),
+        )
+    )
+
+    # If database loading was enabled, start mongodb as well
```

Comment:
Small note here, mongo does not exists in rolling and we will be dropping support for warehouse_ros_mongo. This may stay here for now, but should be switched to `warehouse_ros_sqlite` at some point or stay disabled / commented out.

Reply:
I've never known what the warehouse code does. :shrug:

Based on the documentation [here](https://github.com/ros-planning/warehouse_ros_sqlite) it seems like there should be an added field in `MoveItConfigs`. Does that sound right @JafarAbdi ?